reddit_liveupdate/lib/events.py

```diff
@@ -0,0 +1,103 @@
+from r2.lib.eventcollector import (
+    EventQueue,
+    Event,
+    squelch_exceptions,
+    _datetime_to_millis,
+)
+from r2.lib.utils import sampled
+
+
+class LiveUpdateEvent(Event):
+    def __init__(self, lu_event, **kw):
+        kw.setdefault("topic", "live_thread_events")
+        super(LiveUpdateEvent, self).__init__(**kw)
+
+        self.add("liveevent_id", lu_event._id)
+        self.add("liveevent_fullname", lu_event._fullname)
+        self.add_text("liveevent_title", lu_event.title)
+        self.add_text("liveevent_description", lu_event.description)
+        self.add("liveevent_created_ts", _datetime_to_millis(lu_event._date))
+        if lu_event.banned:
+            self.add_text("liveevent_banned_by", lu_event.banned_by)
+
+        self.add_if_true("liveevent_banned", lu_event.banned)
+        self.add_if_true("liveevent_nsfw", lu_event.nsfw)
+
+    def add_if_true(self, key, value):
+        if value:
+            self.add(key, value)
+
+    @classmethod
+    def get_context_data(cls, request, context):
+        data = super(LiveUpdateEvent, cls).get_context_data(request, context)
+        return data
+
+
+class LiveUpdateEventQueue(EventQueue):
```

Comment:
This is going to open two new handles to the event publisher queues even though `g.events` already has them open.  This class seems to just be a convenient way of getting some functions into the world. Instead of making a new "queue" and putting it on `g`, how about just having some module-level functions in this file and calling them from elsewhere? They can use `g.events` then.